tests/Unit/Evolution/Systems/Cce/BoundaryData.py

```diff
@@ -0,0 +1,240 @@
+                        inverse_spatial_metric):
+    sigma = inverse_spatial_metric[0, :].copy()
+    sigma[0] = cos_phi * sin_theta**2
+    sigma[1] = sin_phi * sin_theta**2
+    sigma[2] = cos_theta * sin_theta
+    norm_of_sigma = math.sqrt(
+        np.einsum("i,j,ij", sigma, sigma, inverse_spatial_metric))
+
+    worldtube_normal = np.einsum("ij,j", inverse_spatial_metric,
+                                 sigma / norm_of_sigma)
+
+    dt_worldtube_normal = np.einsum(
+        "ij,k,jk", 0.5 * np.outer(worldtube_normal, worldtube_normal) -
+        inverse_spatial_metric, worldtube_normal, dt_psi[1:4, 1:4])
+    return dt_worldtube_normal
+
+
+def null_vector_l(dt_worldtube_normal, dt_lapse, dt_psi, dt_shift, lapse, psi,
+                  shift, worldtube_normal):
+    hypersurface_normal_vector = np.pad(worldtube_normal, ((1, 0)),
+                                        'constant').copy()
+    hypersurface_normal_vector[0] = 1.0 / lapse
+    hypersurface_normal_vector[1:4] = -shift / lapse
+    null_l = np.pad(worldtube_normal, ((1, 0)), 'constant').copy()
+    null_l[0] = hypersurface_normal_vector[0] / (
+        lapse - np.einsum("ij,i,j", psi[1:4, 1:4], shift, worldtube_normal))
+    null_l[1:4] = (hypersurface_normal_vector[1:4] + worldtube_normal) / (
+        lapse - np.einsum("ij,i,j", psi[1:4, 1:4], shift, worldtube_normal))
+    return null_l
+
+
+def du_null_vector_l(dt_worldtube_normal, dt_lapse, dt_psi, dt_shift, lapse,
+                     psi, shift, worldtube_normal):
+    hypersurface_normal_vector = np.pad(worldtube_normal, ((1, 0)),
+                                        'constant').copy()
+    hypersurface_normal_vector[0] = 1.0 / lapse
+    hypersurface_normal_vector[1:4] = -shift / lapse
+
+    denominator = (
+        lapse - np.einsum("ij,i,j", psi[1:4, 1:4], shift, worldtube_normal))
+
+    du_hypersurface_normal = np.pad(shift[:], ((1, 0)), 'constant').copy()
+    du_hypersurface_normal[0] = -dt_lapse / lapse**2
+    du_hypersurface_normal[1:4] = - (dt_shift / lapse) \
+        + (np.outer(dt_lapse, shift) / lapse**2)
+
+    du_null_l = (du_hypersurface_normal) / denominator
+    du_null_l[1:4] += (dt_worldtube_normal) / denominator
+    du_denominator = (-dt_lapse + np.einsum(
+        "ij,i,j", dt_psi[1:4, 1:4], shift, worldtube_normal) + np.einsum(
+            "ij,i,j", psi[1:4, 1:4], dt_shift, worldtube_normal) + np.einsum(
+                "ij,i,j", psi[1:4, 1:4], shift, dt_worldtube_normal))
+
+    du_null_l[
+        0] += du_denominator * hypersurface_normal_vector[0] / denominator**2
+    du_null_l[1:4] += du_denominator * (
+        hypersurface_normal_vector[1:4] + worldtube_normal) / denominator**2
+    return du_null_l
+
+
+def dlambda_null_metric(angular_d_null_l, cartesian_to_angular_jacobian, phi,
+                        pi, du_null_l, inverse_null_metric, null_l, psi):
+    dlambda_null_metric = psi.copy()
+    dlambda_null_metric[0, 0] = np.einsum("i,i", null_l[1:4], phi[:, 0, 0]) \
```

Comment:
If you put the whole thing in quotes rather than using `\` then formatters have a better time. E.g. `... = (...)`

I'll let you apply this to other places in this file that also use `\` (I view it as preferable to avoid them because they seem to cause havoc with formatters)